prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt

```diff
@@ -111,10 +132,111 @@ class DeliveryFunction(
             name = "getDeliveryDetails",
             methods = [HttpMethod.GET],
             authLevel = AuthorizationLevel.ANONYMOUS,
-            route = "waters/report/{deliveryId}/delivery"
+            route = "waters/report/{id}/delivery"
+        ) request: HttpRequestMessage<String?>,
+        @BindingName("id") id: String,
+    ): HttpResponseMessage {
+        return this.getDetailedView(request, id)
+    }
+
+    /**
+     * Get a sortable list of delivery facilities
+     *
+     * @param request HTML request body.
+     * @param id Report or Delivery id for the report to get facilities from
+     * @return JSON of the facility list or errors.
+     */
+    @FunctionName("getDeliveryFacilities")
+    fun getDeliveryFacilities(
+        @HttpTrigger(
+            name = "getDeliveryFacilities",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "waters/report/{id}/facilities"
         ) request: HttpRequestMessage<String?>,
-        @BindingName("deliveryId") deliveryId: String,
+        @BindingName("id") id: String,
     ): HttpResponseMessage {
-        return this.getDetailedView(request, deliveryId)
+        try {
+            // Do authentication
+            val authResult = this.authSingleBlocks(request, id)
+            if (authResult != null)
+                return authResult
+
+            val params = HistoryApiParameters(request.queryParameters)
+            val facilityParams = FacilityListApiParameters(request.queryParameters)
+
+            val facilities = deliveryFacade.findDeliveryFacilities(
+                ReportId.fromString(id),
+                params.sortDir,
+                facilityParams.sortColumn,
+            )
+
+            val resultFacilityList: List<Facility> = facilities.map {
+                Facility(
+                    it.testingLabName,
+                    it.location,
+                    it.testingLabClia,
+                    it.positive,
+                    it.countRecords,
+                )
+            }
+
+            return HttpUtilities.okResponse(
```

Comment:
Having multiple returns in a function is not a good practice.  Try to have only one.

Reply:
I was able to combine them